integration-tests/src/test_loop/tests/resharding_v3.rs

```diff
@@ -387,6 +389,67 @@ fn next_block_has_new_shard_layout(epoch_manager: Arc<dyn EpochManagerAdapter>,
         && shard_layout != next_shard_layout
 }
 
+/// Asserts that for each child shard (Mem)Trie and FlatState contains the same final state.
+fn assert_state_sanity_for_children_shard(parent_shard_uid: ShardUId, client: &Client) {
+    let final_head = client.chain.final_head().unwrap();
+
+    for child_shard_uid in client
+        .epoch_manager
+        .get_shard_layout(&final_head.epoch_id)
+        .unwrap()
+        .get_children_shards_uids(parent_shard_uid.shard_id())
+        .unwrap()
+    {
+        let state_root = *client
+            .chain
+            .get_chunk_extra(&final_head.last_block_hash, &child_shard_uid)
+            .unwrap()
+            .state_root();
+
+        let trie = client
+            .runtime_adapter
+            .get_trie_for_shard(
+                child_shard_uid.shard_id(),
+                &final_head.last_block_hash,
+                state_root,
+                false,
+            )
+            .unwrap();
+        let trie_state =
+            trie.lock_for_iter().iter().unwrap().collect::<Result<HashSet<_>, _>>().unwrap();
+
+        let flat_store = client.chain.chain_store().store().flat_store();
+        let flat_store_state = flat_store
+            .iter(child_shard_uid)
+            .map_ok(|(key, value)| {
+                let value = match value {
+                    FlatStateValue::Ref(value) => client
+                        .chain
+                        .chain_store()
+                        .store()
+                        .trie_store()
+                        .get(child_shard_uid, &value.hash)
+                        .unwrap()
+                        .to_vec(),
+                    FlatStateValue::Inlined(data) => data,
+                };
+                (key, value)
+            })
+            .collect::<Result<HashSet<_>, _>>()
+            .unwrap();
+
+        let diff = trie_state.symmetric_difference(&flat_store_state);
+        if diff.clone().count() == 0 {
+            continue;
+        }
+        for (key, value) in diff {
+            tracing::error!(target: "test", key=ALL_COLUMNS_WITH_NAMES[key[0] as usize].1, ?value, "Difference between trie and flat state!");
+        }
+        // TODO: fix nearcore and re-enable the assertion.
+        // assert!(false, "trie and flat state mismatch!");
```

Comment:
Unfortunately it fails even in the most basic scenario:

```
 2.931s ERROR test: Difference between trie and flat state! key="BandwidthSchedulerState" value=[94, 34, 173, 43, 244, 184, 180, 56, 30, 177, 218, 185, 59, 76, 81, 133, 182, 249, 56, 102, 121, 245, 66, 29, 103, 178, 93, 54, 227, 126, 83, 166]
 2.931s ERROR test: Difference between trie and flat state! key="BandwidthSchedulerState" value=[147, 209, 48, 24, 242, 139, 119, 0, 109, 142, 129, 43, 210, 164, 204, 188, 242, 119, 3, 143, 202, 217, 165, 33, 123, 228, 77, 126, 59, 152, 251, 209]
 ```
 
`BANDWIDTH_SCHEDULER_STATE` is duplicated in every child in Flat Storage. I think there's no special handling for it during memtrie split

Reply:
Fixed, it was a mistake in the test.